splink/profile_data.py

```diff
@@ -235,32 +237,33 @@ def profile_columns(
     """
 
     splink_df_dict = db_api.register_multiple_tables(table_or_tables)
+
+    pipeline = CTEPipeline()
+    sql = vertically_concatenate_sql(
+        splink_df_dict, salting_reqiured=False, source_dataset_column_name=None
+    )
+    pipeline.enqueue_sql(sql, "__splink__df_concat")
+
     input_dataframes = list(splink_df_dict.values())
-    input_aliases = list(splink_df_dict.keys())
+
     input_columns = input_dataframes[0].columns_escaped
 
     if not column_expressions:
         column_expressions_raw = input_columns
     else:
         column_expressions_raw = ensure_is_list(column_expressions)
-    column_expressions = expressions_to_sql(column_expressions_raw)
 
-    pipeline = CTEPipeline(input_dataframes)
+    # If the user has provided any ColumnExpression, convert to string
```

Comment:
Lines 256-264  is a separate bug I found in this code whereby ColumnExpressions were not being converted into strings